### PR TITLE
Apply async semantic token modifier to async/await keywords

### PR DESCRIPTION
**Pull request by hi-rustin**
close https://github.com/rust-analyzer/rust-analyzer/issues/8633

### CODE REVIEW

File: crates/ide/src/syntax_highlighting/highlight.rs [hunk 1]
```diff
@@ -255,6 +255,7 @@ pub(super) fn element(
                     })
                     .map(|modifier| h | modifier)
                     .unwrap_or(h),
+                T![async] | T![await] => h | HlMod::Async,
```
Comment (Veykril) [inline crates/ide/src/syntax_highlighting/highlight.rs:258]: `T![await]` is being matched in an earlier arm already as its being tagged with ControlFlow as well, you'll have to special case that one

**Comment by Veykril** [comment]
Could you add some test data to https://github.com/rust-analyzer/rust-analyzer/blob/9803a9a148e1c5850d2f649ffbc1bb3575cfa929/crates/ide/src/syntax_highlighting/tests.rs#L9-L216?

Just an async function definition and a call of the function + awaiting its result should suffice. Doesn't need to to type check.

File: crates/ide/src/syntax_highlighting/highlight.rs (hunk 1 above)
Comment (hi-rustin) [reply crates/ide/src/syntax_highlighting/highlight.rs]: Modified, now only match the async.

**Comment by hi-rustin** [comment]
> Could you add some test data to
> 
> https://github.com/rust-analyzer/rust-analyzer/blob/9803a9a148e1c5850d2f649ffbc1bb3575cfa929/crates/ide/src/syntax_highlighting/tests.rs#L9-L216
> ?
> 
> Just an async function definition and a call of the function + awaiting its result should suffice. Doesn't need to to type check.

Added.

File: crates/ide/src/syntax_highlighting/highlight.rs (hunk 1 above)
Comment (Veykril) [reply crates/ide/src/syntax_highlighting/highlight.rs]: Ah sorry I meant removing `T![await]` from the other arms and adding `T![await] => h | HlMod::Async | HlMod::ControlFlow`, as its both control flow and async.

Comment (hi-rustin) [reply crates/ide/src/syntax_highlighting/highlight.rs]: Sorry, I misunderstood you.

Comment (hi-rustin) [reply crates/ide/src/syntax_highlighting/highlight.rs]: Fixed.

Comment (hi-rustin) [reply crates/ide/src/syntax_highlighting/highlight.rs]: Thanks for your help!

Comment (Veykril) [reply crates/ide/src/syntax_highlighting/highlight.rs]: No problem

**Comment by Veykril** [comment]
Thanks!
bors r+

**Pull request merged**